src/routes/user.js

```diff
@@ -5,9 +5,11 @@ import { UserController } from '../controllers';
 const router = Router();
 
 const { authenticate } = AuthMiddleware;
-const { profileCheck } = UserMiddleware;
-const { updateProfile, getProfile } = UserController;
+const { profileCheck, onDeleteAccount } = UserMiddleware;
+const { updateProfile, getProfile, deleteAccount } = UserController;
 router.put('/profile/:userId', authenticate, profileCheck, updateProfile);
 router.get('/profile/:userId', authenticate, profileCheck, getProfile);
+router.put('/profile/:userId', authenticate, profileCheck, updateProfile);
+router.delete('/account/:userId', authenticate, onDeleteAccount, deleteAccount);
```

Comment:
since this route only deletes a user, its best to reduce the url to just `/v1.0/user/{id}`. it's by default a DELETE call anyways.
```suggestion
router.delete('/:userId', authenticate, onDeleteAccount, deleteAccount);
```